Content.Server/Chat/ChatSystem.cs

```diff
@@ -329,19 +331,76 @@ private IEnumerable<INetChannel> GetDeadChatClients()
             .Select(p => p.ConnectedClient);
     }
 
-    private string SanitizeMessageCapital(EntityUid source, string message)
+    private (string, int) RadioPrefix(EntityUid source, string message)
     {
-        if (message.StartsWith(';'))
+        bool channelMessage = message.StartsWith(':') || message.StartsWith('.');
+        bool radioMessage = message.StartsWith(';') || channelMessage;
+        if (radioMessage)
         {
-            // Special case for ";" messages
+            // Special case for empty messages
             if (message.Length == 1)
-                return "";
-
-            // Remove semicolon
-            message = message.Substring(1).TrimStart();
+                return ("", -1);
 
-            // Capitalize first letter
-            message = message[0].ToString().ToUpper() + message.Remove(0, 1);
+            // Look for a prefix indicating a destination radio channel.
+            int chan = -1;
+            Console.WriteLine(channelMessage + " " + (message.Length >= 2));
+            if (channelMessage && message.Length >= 2)
+            {
+                switch (message[1])
+                {
+                case 'c':
+                    // Command
+                    chan = 1;
+                    break;
+                case 'y':
+                    // CentCom
+                    chan = 2;
+                    break;
+                case 'e':
+                    // Engineering
+                    chan = 3;
+                    break;
+                case 'm':
+                    // Medical
+                    chan = 4;
+                    break;
+                case 'n':
+                    // Science
+                    chan = 5;
+                    break;
+                case 's':
+                    // Security
+                    chan = 6;
+                    break;
+                case 'v':
+                    // Service
+                    chan = 7;
+                    break;
+                case 'u':
+                    // Supply
+                    chan = 8;
+                    break;
+                case 't':
+                    // Syndicate
+                    chan = 9;
+                    break;
+                default:
+                    _popup.PopupEntity(Loc.GetString("chat-manager-no-such-channel"), source, Filter.Entities(source));
+                    chan = -1;
+                    break;
+                }
+
+                // Strip message prefix.
+                var parts = message.Split(' ').ToList();
+                parts.RemoveAt(0);
+                message = string.Join(" ", parts);
+            }
```

Comment:
These should be done in prototypes as using it is cumbersome using this in prototypes as is.

Also don't do console.writelines, use the logger.

Reply:
Console.WriteLines() was left from debugging, oops! It has now been removed.